File: python/checkRepo.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import os

tempFile = "tempFile"

def updateDir(path):
    tempFilePath = "{0}/{1}".format(path, tempFile)

    # write log to a temp file
    os.system("cd {0} && git status > {1}".format(path, tempFile))

    # read it
    fr = open(tempFilePath, "rb")
    content = fr.read()
    fr.close()

    # notify if not up-to-date
    if "up-to-date" not in content:
        print path

    # remove temp file
    os.remove(tempFilePath)

def main():
    for path in os.listdir("./"):
        if os.path.isdir(path):
            updateDir(path)

if __name__ == '__main__':
    main()

File: ruby/checkRepo.rb
#!/env/bin/ruby
# encoding: utf-8

def cdRepo path
  tempFile = "what_ever_things_here"
  tempFilePath = "#{path}/#{tempFile}"
  system "cd #{path} && git status > #{tempFile}"

  fr = open tempFilePath
  content = fr.read
  fr.close

  File.delete tempFilePath

  if not content.include? "up-to-date"
    puts path
  else
    puts "OK"
  end
end

def checkRepo
  Dir.entries("./").each do |path|
    if File.directory? path and not /^\.+$/ =~ path
      cdRepo path
    end
  end

  puts "Check completed..."
end

if __FILE__ == $0
  checkRepo()
end


